redis/client.py

```diff
@@ -358,6 +362,15 @@ def set_response_callback(self, command: str, callback: Callable) -> None:
         """Set a custom Response Callback"""
         self.response_callbacks[command] = callback
 
+    def _cache_invalidation_process(
```

Comment:
Can we add comments on why all of this is necessary. Be kind to our future selves